### PR TITLE
Move 3p AWSNativeSDK ios to github

### PR DESCRIPTION
**Pull request by onecent1101**
## Details
1. Move existing FindAWSNativeSDK to github
2. Add custom build and install script for ios

Note: Bump the ios package version as the FindAWSNativeSDK need a copyright update

## Testing
Tested locally
AR testing https://jenkins-pipeline.agscollab.com/blue/organizations/jenkins/O3DE-LY-FORK/detail/LYN-6779/4/pipeline/

Signed-off-by: liug <liug@amazon.com>

### CODE REVIEW

**Review by lawsonamzn — APPROVED** [review]
Looks good to me.  Isn't this already there?

**Comment by onecent1101** [comment]
> 
> 
> Looks good to me. Isn't this already there?

Next step is to upload rev4 version to dev s3, verify through AR build, promote to prod s3 and send out engine 3rdparty code version update

File: package-system/AWSNativeSDK/FindAWSNativeSDK.cmake.iOS [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+include(CMakeParseArguments)
+
+set(AWSNATIVESDK_PACKAGE_NAME AWSNativeSDK)
+
+set(AWS_BASE_PATH ${CMAKE_CURRENT_LIST_DIR}/${AWSNATIVESDK_PACKAGE_NAME})
+
+# Include Path
+set(AWSNATIVESDK_INCLUDE_PATH ${AWS_BASE_PATH}/include)
+
+# Only static libs are supported for this revision
```
Comment (junbo75) [inline package-system/AWSNativeSDK/FindAWSNativeSDK.cmake.iOS:17]: Curious: Is there a plan to add dynamic libraries later or is it not supported on iOS at all?

File: package-system/AWSNativeSDK/build_AWSNativeSDK_ios.sh [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+    echo "CMake Build $build_type Static to $bld_path/${build_type}_Static"
+    cmake --build "$bld_path/${build_type}_Static" --config $build_type -j 12 || (echo "CMake Build $build_type Static to $bld_path/${build_type}_Static failed" ; exit 1)
+}
+
+make_configure_and_build_curl() {
+  rm -rf "temp/curl"*
+
+  echo "Downloading Curl 7.65.3"
+  (cd temp && curl -o curl-7.65.3.zip "https://curl.se/download/curl-7.65.3.zip") || exit 1
```
Comment (junbo75) [inline package-system/AWSNativeSDK/build_AWSNativeSDK_ios.sh:41]: Why not clone from Github like what your implementation in the cmake_configure_and_build_curl() function below?

File: package-system/AWSNativeSDK/FindAWSNativeSDK.cmake.iOS (hunk 1 above)
Comment (onecent1101) [reply package-system/AWSNativeSDK/FindAWSNativeSDK.cmake.iOS:17]: no unless there is strong requirement, as that is how we setup previously (not sure whether there is any limitation here)

File: package-system/AWSNativeSDK/build_AWSNativeSDK_ios.sh (hunk 2 above)
Comment (onecent1101) [reply package-system/AWSNativeSDK/build_AWSNativeSDK_ios.sh:41]: source cloned from github repo doesn't support make build directly.

**Review by junbo75 — APPROVED** [review]

File: package-system/AWSNativeSDK/build_AWSNativeSDK_ios.sh [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+  (cd temp/curl-7.65.3 && ./configure --disable-shared --enable-static --enable-ipv6 --with-secure-transport --host="arm-apple-darwin" --prefix=$(pwd)/../curl_install) || exit 1
+  (cd temp/curl-7.65.3 && make) || exit 1
+  (cd temp/curl-7.65.3 && make install) || exit 1
+
+  export CFLAGS=$EXISTING_CFLAGS
+  EXISTING_CFLAGS=
+}
+
+# TODO: curl cmake is poorly maintained by community, switch to cmake when it is ready
```
Comment (lmbr-pip) [inline package-system/AWSNativeSDK/build_AWSNativeSDK_ios.sh:57]: This is an unsubstantiated comment, theres no justification for this statement given. Also we are using CMake so unsure what task actually is. How does anyone reading this take action? What does 'poorly maintained' mean in this context, what is the cmake action?

**Review by lmbr-pip — COMMENTED** [review]
LG

IMHO 1 comment needs an update.

File: package-system/AWSNativeSDK/build_AWSNativeSDK_ios.sh (hunk 3 above)
Comment (onecent1101) [reply package-system/AWSNativeSDK/build_AWSNativeSDK_ios.sh:57]: actually we are not using cmake for curl build, instead we are using makefile.
the comment "cmake is poorly maintained by community" is copied from cmake build result, as cmake failed when I tried to use it to build.
I can remove the whole function, left it to have a reference we can use later.

**Review by lawsonamzn — APPROVED** [review]

**Pull request merged**